### ISSUE TITLE
Onvif camera with two streams

### ISSUE THREAD

**Issue by Barney1977** [AdyRock/com.onvif#44]
Hi Adrian,
I installed 2 extra Axis devices. both are the same type P3715PLVE. this camera exist of two separated objectives at the same network-connection. When I tried to add one of those as onvif camera I get a failure as soon I choose the channel to connect.
first step - searching device 
next step - add username/password and portsettings
next step - choose channel  (1 = objective 1, 2 = objective 2, 3 is merge of both pictures )

This step ends what ever choice I make with te same message: invalid string type! See attachment.. Is this something I can solve by my own?

![onvif II](https://user-images.githubusercontent.com/121728227/215753796-fce65654-2069-401f-892a-515eef70bd39.jpg)
![onvif I](https://user-images.githubusercontent.com/121728227/215753816-58c83e53-f1eb-4d99-acf4-b3553a7fef9f.jpg)

**Comment by AdyRock** [comment]
Could you send me the log from the Configure App screen so I can hopefully see what is causing the error

**Comment by Barney1977** [comment]
I just sent it. Op 1 feb. 2023 om 18:26 heeft Adrian Rockall ***@***.***> het volgende geschreven:﻿
Could you send me the log from the Configure App screen so I can hopefully see what is causing the error

**Comment by Barney1977** [comment]
Hi adrian,Since wednesday app stops working frequently again. It has been fine for more then a Month before. I sent a diagnostic report Code: a597624b-3340-44bf-b87b-e81be1ffea2dMet vriendelijke groet, Henk van Barneveld  Op 1 feb. 2023 om 19:05 heeft Henk van Barneveld ***@***.***> het volgende geschreven:﻿I just sent it. Op 1 feb. 2023 om 18:26 heeft Adrian Rockall ***@***.***> het volgende geschreven:﻿
Could you send me the log from the Configure App screen so I can hopefully see what is causing the error

**Comment by Barney1977** [comment]
Hi adrian,I think i know what cause pauze mode of the onvif app. As soon i set Motiondetection active in homey device page ( see attachment) it goes wrong. Activating Motion by flow seems to works well. Met vriendelijke groet, Henk van Barneveld   Op 1 feb. 2023 om 18:26 heeft Adrian Rockall ***@***.***> het volgende geschreven:﻿
Could you send me the log from the Configure App screen so I can hopefully see what is causing the error

**Comment by AdyRock** [comment]
Unfortunately GitHub doesn't support attachments via email so you have to login and add them to the post.
So I can see what is causing the error, I need to the log from Configure App screen. There is a button there to send the log.

**Comment by Barney1977** [comment]
Hi Adrian, please tell me how i can sent logging by email-Github. I only can choose in app for sending log to developer. I already did it.. After installing most recent experimental app sending log caused a crash. I have to restart homey app to get in again.
I can share the faults as printscreen. 
![onvif faults](https://user-images.githubusercontent.com/121728227/220589519-f221568f-5428-495f-815d-24b3549260e5.jpg)

**Comment by AdyRock** [comment]
I have received several logs today from the same user, but I can't identify users from the log, so not sure if they are yours. Is one of your cameras named "Kantoor Duke"?

When you look at the log in the app, does it show the error you see in red above?

**Comment by Barney1977** [comment]
Yes. Thats right. And faults about that camera are Caused by turn off power Supply if somebody is at homeMet vriendelijke groet, Henk van Barneveld  Op 22 feb. 2023 om 15:44 heeft Adrian Rockall ***@***.***> het volgende geschreven:﻿
I have received several logs today from the same user, but I can't identify users from the log, so not sure if they are yours. Is one of your cameras named "Kantoor Duke"?
When you look at the log in the app, does it show the error you see in red above?

**Comment by Barney1977** [comment]
I do not received a part of log with something red in it…Met vriendelijke groet, Henk van Barneveld   Op 22 feb. 2023 om 15:44 heeft Adrian Rockall ***@***.***> het volgende geschreven:﻿
I have received several logs today from the same user, but I can't identify users from the log, so not sure if they are yours. Is one of your cameras named "Kantoor Duke"?
When you look at the log in the app, does it show the error you see in red above?

**Comment by AdyRock** [comment]
In the log page, could you set the level to Basic Information so it reduces the amount of data. Then clear the log and try to add the cameras again. As soon as you see an error, go back to the log page and send it.

**Comment by Barney1977** [comment]
Hi adrian,I tried…but since i installed the pre-release .30 the app crashed constantly. I’am not able to add a cameraMet vriendelijke groet, Henk van ***@***.***:www.nsecure.nl Op 22 feb. 2023 om 16:44 heeft Adrian Rockall ***@***.***> het volgende geschreven:﻿
In the log page, could you set the level to Basic Information so it reduces the amount of data. Then clear the log and try to add the cameras again. As soon as you see an error, go back to the log page and send it.

**Comment by Barney1977** [comment]
Done,if right you just received the basic log Met vriendelijke groet, Henk van ***@***.***:www.nsecure.nl Op 22 feb. 2023 om 16:44 heeft Adrian Rockall ***@***.***> het volgende geschreven:﻿
In the log page, could you set the level to Basic Information so it reduces the amount of data. Then clear the log and try to add the cameras again. As soon as you see an error, go back to the log page and send it.

**Comment by AdyRock** [comment]
There is a new test version that might solve the crashing.

**Comment by Barney1977** [comment]
Thx,Installeer it directly.Kind Regards, Henk van Barneveld  Op 31 mrt. 2023 om 12:35 heeft Adrian Rockall ***@***.***> het volgende geschreven:﻿
There is a new test version that might solve the crashing.

**Comment by Barney1977** [comment]
Hi Adrian,
I'am wondering or it's already cleared or it's possible to connect both streams of a dual stream camera to Homey already.
For me it seems it's still not possible. when searching for new devices I find all different streams but i'am not able to finish integration.

**Comment by AdyRock** [comment]
To be honest I can't remember. 
Could you set the Diagnostics log to Detailed, add the camera and then send the log when you have gone as far as you can.

**Comment by Barney1977** [comment]
I really Think I’am going crazy.I choose requested log-level. Tried to install camera and it works…. I only had to adjust setting (camera instead of sensor ) and it works. Sorry, and Thanks. I really do not know what went wrong yesterday.Kind regards,Henk van Barneveld Op 9 jan 2024 om 09:57 heeft Adrian Rockall ***@***.***> het volgende geschreven:﻿
To be honest I can't remember.
Could you set the Diagnostics log to Detailed, add the camera and then send the log when you have gone as far as you can.

**Comment by AdyRock** [comment]
That's great news, thanks for letting me know.

**Comment by Barney1977** [comment]
Hi Adrian,May I ask another question about onvif app?Because of high memory use of my homey pro 2023 i checked all apps seperately. I notice That onvif is using a lot all the time. >60mb rss and >50mb pss as minimum. Can i adjust/configure it to reduce it?Kind regards,Henk van barneveldOp 9 jan 2024 om 19:24 heeft Adrian Rockall ***@***.***> het volgende geschreven:﻿
That's great news, thanks for letting me know.

**Comment by AdyRock** [comment]
The only thing that might make a difference is the logging. Make sure that is set to errors only. Apart from that I have no idea why it would use so much, but I will investigate on my HP2023. 
On my HP2019 it is only using 5MB of storage and 28MB of memory.
